staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +107,64 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public and private key files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	ok := false
+	var keys []interface{}
+	for len(keyData) > 0 {
+		var block *pem.Block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+		// Only use PEM "PUBLIC KEY" or "PRIVATE KEY" blocks without extra headers
+		if block.Type != PublicKeyBlockType || block.Type != PrivateKeyBlockType || len(block.Headers) != 0 {
+			continue
+		}
+
+		switch block.Type {
+		case ECPrivateKeyBlockType:
+			privateKey, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case RSAPrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case PrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			if parsedRSAKey, parseSuccess := privateKey.(*rsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedRSAKey.PublicKey)
+			} else if parsedECKey, parseSuccess := privateKey.(*ecdsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedECKey.PublicKey)
+			} else {
+				return keys, err
+			}
```

Comment:
I don't see any tests in the code that specifically call a `BEGIN RSA PUBLIC KEY` block, even in the jwt code that serviceaccount uses (https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/dgrijalva/jwt-go/rsa_utils.go#L42-L69). If I pass a `BEGIN RSA PUBLIC KEY` using the old PKCS1 format for public keys, ParsePKIXPublicKey errors out on the block. The only extra thing that I found was that it will also try to parse public keys out of certificates, so I just added that.

Is there somewhere I'm missing that the code calls a `BEGIN RSA PUBLIC KEY` block instead of a `BEGIN PUBLIC KEY` block?